### ISSUE TITLE
Can't set value to form 

### ISSUE THREAD

**Issue by CristianCucunuba** [DairyStateDesigns/reacto-form#11]
If I try to set the value object on the form, the form starts to re-render infinitely and blocks the inputs 

Repo: https://codesandbox.io/s/exciting-turing-146t8?file=/src/Form.js

**Comment by CristianCucunuba** [comment]
Only happens with variables declare inside the component

**Comment by aldeed** [comment]
Yeah, I'm not sure this would really be fixable because that's how React works. If you create a new object every time the parent component renders, then it will always be interpreted as a new/changed `value`. In a real-world implementation, your `value` would always be an incoming prop or would be from `useState`, so this never happens.

The only workaround I can imagine would be to do a deep comparison of the `value` object. I'll leave this issue open in case that can be done, but deep compare is pretty tricky to get right in JS.
